src/core/transform.js

```diff
@@ -25,12 +25,37 @@ import p5 from './main';
  * alt="The transformation matrix used when applyMatrix is called"/>
  *
  * @method applyMatrix
- * @param  {Number|Array} a numbers which define the 2×3 matrix to be multiplied, or an array of numbers
- * @param  {Number} b numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} c numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} d numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} e numbers which define the 2×3 matrix to be multiplied
- * @param  {Number} f numbers which define the 2×3 matrix to be multiplied
+ * @param  {Array} arr an array of numbers
+ * @chainable
+ */
+/**
+ * @method applyMatrix
+ * @param  {Number} a_2x3 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} b_2x3 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} c_2x3 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} d_2x3 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} e_2x3 numbers which define the 2×3 matrix to be multiplied
+ * @param  {Number} f_2x3 numbers which define the 2×3 matrix to be multiplied
```

Comment:
Why is the parameter renamed here from the original `a, b, c,...`?

Reply:
The YuiDoc generator is having problems when the params of overloads have the same name.  Would it be better to be all together?

```
  * @param  {Number} a numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} b numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} c numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} d numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} e numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} f numbers which define the 2×3 or 4x4 matrix to be multiplied
  * @param  {Number} [g] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [h] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [i] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [j] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [k] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [l] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [m] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [n] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [o] numbers which define the 4x4 matrix to be multiplied
  * @param  {Number} [p] numbers which define the 4x4 matrix to be multiplied
  ```

Reply:
What are the problems it is having? As we have other functions with overloads using same parameter names elsewhere and they work fine for the most part (`loadJSON()` for example).

Reply:
@limzykenneth I shall investigate further sir :)

Reply:
Apparently it breaks when you try to give descriptions to the overloaded params, the `loadJSON()` example does not have descriptions for the params.

Shall I combine all variables into a single method with optional params?  Or should I create better variable names and 
have 3 distinct methods (I have committed a new file change that has nicer names) let me know what you think I should do :)

Reply:
`loadJSON()` does have descriptions for each of the parameters. It probably clearer to use the reference page as a guide instead of the inline docs.

The descriptions can only be defined once for each parameter meaning the parameter with the same name cannot have two different descriptions (which make sense when we look at the layout of the reference page, all parameters are listed and described together). If the overload have new parameters that hasn't been described, they can have descriptions as well (see `tint()` for this case).

Reply:
> The descriptions can only be defined once for each parameter meaning the parameter with the same name cannot have two different descriptions (which make sense when we look at the layout of the reference page, all parameters are listed and described together). If the overload have new parameters that hasn't been described, they can have descriptions as well (see `tint()` for this case).

This is correct.  What shall we do in this case?  

Option 1) Overload method has different param names and thus different param descriptions (as with tint())
Option 2) Overload for 4x4 matrix has new descriptions for param g - p
Option 3) Remove param descriptions for both overloads that access 6 or 16 values and add to overall method description

Any advice on what is needed to get this PR in would be appreciated :)

Reply:
In this case only a-f would present overlap of names with other overloads which means that these will need one description each that covers when they are used for 2x3 or 2x4 cases, ie `numbers which define the 2×3 or 4x4 matrix to be multiplied`. g-p are only used in 4x4 so they can have a 4x4 specific description ie `numbers which define the 4x4 matrix to be multiplied`. Finally array is its own case so can have its own description, ie `arr an array of numbers - should be 6 or 16 length (2*3 or 4*4 matrix values)`.

Which is to say for descriptions we are close here already, you just need to add the description back in instead of leaving them completely blank or changing their names.